docs: add a contributing FAQs document

aayush0325 opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   adds an initial doc with frequently asked questions by first time contributors

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aayush0325 committed:
docs: add a contributing FAQs document

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented:
we need to add a question for how to get past the uninitialized variable linting error in C benchmarks as faced in numerous packages in `stats/strided/*`

aayush0325 commented:
kindly check if we need to add more make commands other than the ones already present in the doc, i was conflicted about whether to add one for generating polynomials via `evalpoly.js`

kgryte commented:
@anandkaranubc I'm curious if you have any additional thoughts on FAQs that we should include given that you've also been doing a fair amount of code review.

aayush0325 commented:
not sure if i've taken the correct approach for adding images in this doc, currently i've just copy pasted the screenshots, this way the images are visible when we open the doc on a browser but if i try to preview this on VS Code the images are not showing up

kgryte commented:
Also, @Planeshifter and @headlessNode may have some thoughts.

aayush0325 committed:
docs: remove todo and add a FAQ for uninit var suppression

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

saurabhraghuvanshii commented:
It is good if sync with upstream with rebase also include because maximum contributer faces issue with rebase and in result they commit unwanted commits to file

aayush0325 commented:
>It is good if sync with upstream with rebase also include because maximum contributer faces issue with rebase and in result they commit unwanted commits to file

Thanks for pointing it out @saurabhraghuvanshii! this one was on my todolist for this doc, writing it soon

headlessNode commented:
Although this is not a FAQ. This suggestion, if accepted, could be implemented in a separate PR.

One thing I've noticed regarding questions asked on the Gitter channel is that people usually post a screenshot along with a vague question like *"How do I solve this error?"* 

We could encourage a format which may include:

- Description (what they are trying to do, maybe a link to a PR)
- Issue encountered
- Steps taken (what they’ve tried so far)

Having a structured questions format could provide several benefits:

- People willing to help will quickly understand the issue resulting in little to no back-and-forth clarifications.
- When describing their issue in detail, people may come upon a solution on their own.

Would love to hear your thoughts on this @Planeshifter @kgryte

kgryte commented:
@headlessNode Totally agree. Rarely do folks on Gitter provide enough info to help debug, resulting in a series of back and forths which is not a productive use of time.

aayush0325 commented:
adding this soon, thank you for your suggestion @headlessNode!

i plan to keep the format like this:

question: ---description of issue----
steps to reproduce: (if applicable)
error message: error message / images / logs / diverging results from expected values 
current approach: --how the person has tried solving this---
reference: ---a PR link (if applicable)---

please feel free to suggest changes in this

headlessNode commented:
@aayush0325 Great. We can iterate on this when you open a PR. 🎆

aayush0325 commented:
> @aayush0325 Great. We can iterate on this when you open a PR. 🎆 

I was planning to add it in this doc itself😅. would that be okay

headlessNode commented:
IMO, it should be a separate document and we could link it in this doc like:

If you have a question that isn’t covered here, Feel free to reach out on our Gitter channel and make sure to follow our guide(Link) on how to ask questions.

@kgryte what do you think?

kgryte commented:
It's a bit similar to our issue template: https://github.com/stdlib-js/stdlib/blob/develop/.github/ISSUE_TEMPLATE/bug_report.yml

Something for inspiration: https://stackoverflow.com/help/minimal-reproducible-example

I think we could probably add something similar for stdlib in the `docs/` folder, and given that it might be a bit longer, I think it fine to place in a separate document. Whether that document is added in this PR or in a separate PR doesn't matter to me. Either is fine.

aayush0325 commented:
> If you have a question that isn’t covered here, Feel free to reach out on our Gitter channel and make sure to follow our guide on how to ask questions.

makes sense to me, probably gonna raise another PR for that

anandkaranubc commented:
@aayush0325 This is definitely a well-thought-out FAQ document!

Here’s my feedback:

1. We can add a list of questions (as links) at the top for better navigation, similar to [this format](https://github.com/stdlib-js/stdlib/blob/6d9ed107bc5e674421c5773619a34d5c0fe5a96a/FAQ.md#frequently-asked-questions).
   
2. A commonly asked question on the Gitter channel is: "I am a first-time contributor to stdlib, where should I start?" This question could be placed right after the introduction.

3. The section on setting up the development environment is focused only on the devcontainer setup, which might be misleading. It would be better to present two options:  
   - Manually setting up the development environment  
   - Setting up the devcontainer  

   Also, the devcontainer does not support ARM64 architectures yet (this is on my to-do list). We can mention [this issue](https://github.com/stdlib-js/stdlib/issues/4934) for those interested in learning more or contributing to it.

4. The devcontainer setup section takes up a large portion of the document. A suggestion: We can move it to a separate document and link to it from the FAQ. In fact, this can be done for both manual and devcontainer setup instructions.

5. For the question on "Why do JS and C return different values?", we can add these references:  
   - [Discussion link](https://github.com/stdlib-js/stdlib/pull/2298#discussion_r1624765205)  
   - [Reference comment](https://github.com/stdlib-js/stdlib/blob/1f9cb760e3345cc7e08320a11f6a051873ef3586/lib/node_modules/%40stdlib/math/base/special/spence/test/test.native.js#L90)  

6. The weekly office hours meeting link can be updated to the [issues section](https://github.com/stdlib-js/meetings/issues), where meeting details are posted.

7. For the question on "How do I generate test fixtures?", we can also mention:  
   - `make test-cov` and `make view-cov`  
   - [This link](https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib/test-cov) for checking 100% test coverage locally.  

8. The sample `manifest.json` content can be shortened to only include the `"confs"` section with a starting line like `// Other sections above...`. This would make it clearer for the reader on what's being pointed out.

9. In the Make commands section:  
   - "Generate Julia Fixtures" → Rename to "Generate Test Fixtures" and list Python, Julia, and C++ as "some" examples, and add the below link for more.  
     - Link: [Test Fixtures](https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib/test-fixtures).  
   - "Run Examples" → Same as above. List make commands for running C and JS examples and add the below link.
     - Link: [Examples](https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib/examples).  
   - "Run Benchmarks" → Same and add a link to [Benchmarks](https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib/benchmark).  

10. We can add an "Other Links" section for useful resources:  
    - [Style guides](https://github.com/stdlib-js/stdlib/tree/develop/docs/style-guides) (covers variable declaration, initialization, "negative" spacing, etc.)  
    - ["All" available make commands](https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib).  

Other Popular Questions to Add (I understand that not all questions can be included due to length constraints and to keep the document readable):

1. "How should I name my PR?"  
   - Short answer: Look at similar PRs that have already been merged.  

2. "How do I call the stdlib bot on my PR?"  
   - This can also include a note on "Understanding PR lint failures".  

**Phew! Open to everyone’s feedback and happy to provide draft answers for the above.**

kgryte commented:
Another thing to add: when to use decimals and when not to use decimals. E.g., https://github.com/stdlib-js/stdlib/pull/5353/files#r1966326512.

aayush0325 committed:
docs: add devcontainter setup in a separate doc

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented:
A related document that we have can be found here: https://github.com/stdlib-js/stdlib/blob/develop/SUPPORT.md. This also touches on asking questions and seeking help.

aayush0325 committed:
docs: add content following up to the suggestions of reviewers

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 committed:
docs: update devcontainer setup

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 committed:
docs: add todo

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 committed:
docs: update dev faqs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented:
@aayush0325 Would you mind moving the docs into the `docs/contributing` folder? And also,

- `dev_faqs.md` => `contributing/FAQ.md`
- `devcontainer_setup.md` => `contributing/setting_up_a_devcontainer.md`

For images, such as screenshots, let's add them to a `contributing/img` folder.

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into contributing-faqs-doc

aayush0325 committed:
docs: move docs to docs/contributing/

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 committed:
docs: add required images to docs/contributing/img

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 committed:
docs: link to the images in docs/contributing/img

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on docs/contributing/setting_up_a_devcontainer.md:
```diff
@@ -0,0 +1,140 @@
+
+<div class="image" align="center">
+    <a title="Step 1">
+        <img width=“480” src="./img/step1.png" alt="Dev container setup tutorial">
```
```suggestion
        <img width="480" src="./img/step1.png" alt="Dev container setup tutorial">
```

kgryte committed:
docs: remove fancy quotes

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on docs/contributing/setting_up_a_devcontainer.md:
```diff
@@ -0,0 +1,140 @@
+
+<div class="image" align="center">
+    <a title="Step 2">
+        <img width=“480” src="./img/step2.png" alt="Dev container setup tutorial">
```
You need to remove fancy quotes from each of the images below.

```suggestion
        <img width="480" src="./img/step2.png" alt="Dev container setup tutorial">
```

aayush0325 replied:
on it

kgryte commented on docs/contributing/setting_up_a_devcontainer.md:
```diff
@@ -0,0 +1,140 @@
+
+
+<div class="image" align="center">
+    <a title="Step 1">
```
You can remove these anchor tags, as you aren't using them in this document.

kgryte commented on docs/contributing/setting_up_a_devcontainer.md:
```diff
@@ -0,0 +1,140 @@
+
+<div class="image" align="center">
+    <a title="Step 1">
+        <img width="480" src="./img/step1.png" alt="Dev container setup tutorial">
```
These alt-text description should be more descriptive. For those with screen readers, every image will read as if they are the same thing.

kgryte replied:
See, e.g., https://accessibility.huit.harvard.edu/describe-content-images

kgryte commented on docs/contributing/setting_up_a_devcontainer.md:
```diff
@@ -0,0 +1,140 @@
+
+# Devcontainer Setup
+
+> Step-by-Step Tutorial on How to Set Up the Repository in a Dev Container
```
```suggestion
> Step-by-step tutorial on how to set up a stdlib development environment in a dev container.
```

kgryte commented on docs/contributing/setting_up_a_devcontainer.md:
```diff
@@ -0,0 +1,140 @@
+
+We appreciate your interest in contributing to stdlib! Below, we've provided a step-by-step tutorial on how to set up the project locally on your device using a dev container.
+
+The stdlib repository includes a preconfigured dev container, making it the easiest way to set up your development environment. It ensures proper linting, EditorConfig, and tooling are configured right from the start.
```
We should probably describe what, exactly, a dev container issue. A few sentences should suffice.

kgryte commented on docs/contributing/img/step1.png:
As these images will eventually be joined by other images which are used in other documents, I suggest making these file names more descriptive (e.g., `vscode_dev_container_prompt.png`, etc).

aayush0325 committed:
chore: address code review comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
docs: update dev container description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on docs/contributing/FAQ.md:
```diff
@@ -0,0 +1,391 @@
+
+# Contributing FAQs
+
+> Frequently Asked Questions (FAQs) by First-Time Contributors to stdlib
```
```suggestion
> Frequently Asked Questions (FAQs) by First-Time Contributors to stdlib.
```

aayush0325 committed:
chore: add punctuation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte reviewed (approved):
I think this is good enough for now. We can go ahead and get this in and clean-up in follow-up commits/PRs.

aayush0325 commented:
thanks @anandkaranubc for helping out in this!!

anandkaranubc commented:
@aayush0325 Anytime! Time to spread the word on the Gitter channel!

@kgryte, I wonder if the [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/contributing/development.md) should also be updated to include a reference/link to the FAQ document. This could help connect it directly to the repository documentation.
